src/write-archive/index.ts

```diff
@@ -75,6 +83,54 @@ export async function writeTestResult(
   }
 }
 
+/**
+ * Takes a DOM snapshot, which is either a `Buffer` or an object in json form, and maps all `src` attributes that are equivalent to
+ * one of the entries in the `sourceMap` to the resulting value.
+ *
+ * @param domSnapshot The DOM snapshot upon which to run the mapping. On initial run, this will be a `Buffer`.
+ *                    On recursive runs, this will be a json object.
+ * @param sourceMap A mapping of `string` objects to other `string` objects. All `src` attributes that are keys in this map will be
+ *                  adjusted to be the resulting value.
+ * @returns A JSON string representing the mapped DOM snapshot.
+ */
+async function mapSourceEntries(domSnapshot: Buffer, sourceMap: Map<string, string>) {
+  let jsonBuffer: any;
+  if (Buffer.isBuffer(domSnapshot)) {
+    const bufferAsString = domSnapshot.toString('utf-8');
+
+    // Try to parse as JSON. Our tests don't always return JSON, so this is kind of a hack
+    // to avoid a situation where JSON is expected but not actually given.
+    try {
+      jsonBuffer = JSON.parse(bufferAsString);
+    } catch (err) {
+      return domSnapshot;
+    }
+  } else {
+    jsonBuffer = domSnapshot;
+  }
```

Comment:
Should we just make our tests return JSON then?

Reply:
We could do that, but, I feel like that's a slightly bigger change. Perhaps we could refactor it as part of another ticket, possibly during the cooldown cycle?